Templates/JenkinsPipeline/README.md

```diff
@@ -0,0 +1,132 @@
+* [release pipeline](https://ibm.github.io/z-devops-acceleration-program/docs/branching-model-supporting-pipeline#the-release-pipeline-with-build-packaging-and-deploy-stages) to build, package and deploy a release via UCD
+
+The `Pipeline setup` stage within the [template](Jenkinsfile#L165) configures the subsequent stages, including flags used as conditions when stages and jobs are executed. Please review this stage.
+
+This template is implementing the recommended [Git branching model for mainframe development](https://ibm.github.io/z-devops-acceleration-program/docs/git-branching-model-for-mainframe-dev). If you haven't done yet, we recommend to get familiar with the concepts described in this documentation.
+
+### Pipeline variables
+
+In a default setup, the feature and basic pipelines are triggered for each new commit as soon as the Multibranch integration detects the changes.
+
+The pipeline allows to override the values of the below variables when manually requesting the pipeline. This is especially useful when the application team want to create a release candidate package for higher test environments and production.
+
+Parameter | Description
+--- | ---
+pipelineType     | Pipeline type - either build, release or preview. (Default: build)
+verbose          | Boolean flag to control logging of build framework. (Default: false)
+<!-- releaseType      | Release type - major, minor, patch as input to compute the release version and to set the release candidate and release git tags. (Default: patch) //-->
+
+### Feature Branch pipeline
+
+The pipeline for feature branches executes the following steps:
+
+* Setup
+* Clone
+* Build
+* SonarQube Analysis
+* Create UCD Component version
+
+To run zAppBuild in preview mode, this pipeline can be manually requested with setting the *pipelineType* variable to `preview`.
+
+Overview of the pipeline:  
+
+![Jenkins Feature Branch Pipeline](images/jenkins-pipeline-feature-branch.png)
+
+Pipelines for Feature, Epic and Release Maintenance branches perform the same steps like a feature branch pipeline. 
+
+### Basic build pipeline when merging into the main branch
+
+The basic build pipeline for the `main` branch contains the following stages:
+
+* Setup
+* Clone
+* Build (with TEST options)
+* SonarQube Analysis
+* Create UCD Component version
+* Request UCD Deployment to the integration test environment
+
+It runs automatically when there is a new commit to a repository. `dbbBuild.sh` (from the Common Backend Scripts) will automatically extract the Git tag based on the information in the [baselineReference.config](../Common-Backend-Scripts/samples/baselineReference.config). The `baselineReference.config` file is expected to be maintained in the application's git repository.
```

Comment:
- Should this say: "It runs automatically when there is a new commit to a repository's `main` branch."
- Change "application's git repository" to "application's Git repository".